feature: add floating window footer (#24739)
neovim/neovim: runtime/doc/news.txt

• Improved messages for type errors in `vim.api.*` calls (including `opts` params)

> • Floating windows can now show footer with new `footer` and `footer_pos`
>   config fields.
> 
> • New highlight group |hl-FloatFooter| to highlight floating window footer
>   separately from title.

Reviewer: IMO these don't need to be separate.

```suggestion
• Floating windows can now show a footer with the new `footer` and `footer_pos`
  config fields. |hl-FloatFooter|
```
